subsys/ipc/ipc_service/lib/icmsg.c

```diff
@@ -159,31 +162,47 @@ static void submit_work_if_buffer_free_and_data_available(
 
 static void mbox_callback_process(struct k_work *item)
 {
+	char *rx_buffer;
 	struct icmsg_data_t *dev_data = CONTAINER_OF(item, struct icmsg_data_t, mbox_work);
 
 	atomic_t state = atomic_get(&dev_data->state);
-	int len = spsc_pbuf_read(dev_data->rx_ib, dev_data->rx_buffer,
-				 RX_BUF_SIZE);
 
-	__ASSERT_NO_MSG(len <= RX_BUF_SIZE);
+	uint16_t len = spsc_pbuf_claim(dev_data->rx_ib, &rx_buffer);
```

Comment:
When i first saw `spsc_pbuf_claim` i thaught this cannot be done, as the user could call https://github.com/zephyrproject-rtos/zephyr/blob/df0e03c5512c03f4e3e641cc37641abb1f76082b/include/zephyr/ipc/ipc_service_backend.h#L199 where surely `spsc_pbuf_claim` is called, well - its not. This surprises me, i will review this later just tossing this out out as it seems odd we have different mechanism for claiming rx buffer when no copy rx is used.